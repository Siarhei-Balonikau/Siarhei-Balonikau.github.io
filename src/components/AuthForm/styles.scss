.auth-form {
  &__field {
    & + .auth-form {
      &__field {
        padding-top: 20px;
      }
    }
  }
  
  &__button {
    text-align: center;
    padding-top: 30px;
  }
}