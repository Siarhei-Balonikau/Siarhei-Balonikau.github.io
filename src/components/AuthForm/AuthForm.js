import React from 'react';
import PropTypes from 'prop-types';
import TextField from './../TextField/TextField';
import Button from './../Button/Button';
import './styles.scss';
import { Router, Route, Link } from "react-router-dom";

const AuthForm = (props) => {
  return (
    <form className="auth-form" onSubmit={(e) => props.onSubmit(e)}>
      <div className="auth-form__field">
        Login<br />
        <TextField onChange={props.onChange} field={'login'} value={props.form.login}/>
      </div>
      
      <div className="auth-form__field">
        Password<br />
        <TextField onChange={props.onChange} field={'pass'} value={props.form.pass}/>
      </div>
      
      <div className="auth-form__button">
        <Button name="SUBMIT" />
      </div>
    </form>
  );
}

AuthForm.propTypes = {
  onSubmit: PropTypes.func,
  onChange: PropTypes.func,
  form: PropTypes.object,
};

export default AuthForm;