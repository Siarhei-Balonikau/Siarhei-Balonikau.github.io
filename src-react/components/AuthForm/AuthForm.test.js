import React, { Component } from 'react';
import { shallow } from 'enzyme';
import AuthForm from './AuthForm'; 
import toJson from 'enzyme-to-json'

describe('subscribe component', () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      form: {
        login: 'test'
      }
    };
  
    wrapper = shallow(<AuthForm {...props} />);
  });
  
  it('should render component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});