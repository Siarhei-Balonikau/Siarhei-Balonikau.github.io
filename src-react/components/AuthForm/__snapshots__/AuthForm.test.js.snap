// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe component should render component 1`] = `
<form
  className="auth-form"
  onSubmit={[Function]}
>
  <div
    className="auth-form__field"
  >
    Login
    <br />
    <TextField
      field="login"
      value="test"
    />
  </div>
  <div
    className="auth-form__field"
  >
    Password
    <br />
    <TextField
      field="pass"
    />
  </div>
  <div
    className="auth-form__button"
  >
    <Button
      name="SUBMIT"
    />
  </div>
</form>
`;
